wagtail/migrations/0084_fix_log_entry_json_timestamps.py

```diff
@@ -0,0 +1,129 @@
+# Generated by Django 3.2.6 on 2022-11-09 07:50
+
+import datetime
+
+import django.core.serializers.json
+from django.conf import settings
+from django.db import migrations, models
+from django.utils import timezone, dateparse
+
+
+def legacy_to_iso_format(date_string, tz=None):
+    dt = datetime.datetime.strptime(date_string, "%d %b %Y %H:%M")
+    if settings.USE_TZ:
+        dt = timezone.make_aware(dt, datetime.timezone.utc if tz is None else tz)
+        dt = timezone.localtime(dt, datetime.timezone.utc)
+    # We return the datetime object, so DjangoJSONEncoder will serialize it accordingly.
+    return dt
+
+
+def iso_to_legacy_format(date_string, tz=None):
+    dt = dateparse.parse_datetime(date_string)
+    if dt is None:
+        raise ValueError("date isn't well formatted")
+    if settings.USE_TZ:
+        dt = timezone.localtime(dt, datetime.timezone.utc if tz is None else tz)
+    return dt.strftime("%d %b %Y %H:%M")
+
+
+def migrate_logs_with_created_only(model, converter):
+    for item in model.objects.filter(
+        action__in=["wagtail.revert", "wagtail.rename", "wagtail.publish"]
+    ).only("data"):
+        try:
+            # If a previous_revision was available, the data contains "revision" with
+            # its created date.
+            # Also, there are "wagtail.publish" logs, which don't set data at all.
+            created = item.data["revision"]["created"]
+            # "created" is set to the previous revision's created_at, which is set
+            # to UTC by django.
+            item.data["revision"]["created"] = converter(created)
+        except ValueError:
+            # TODO TBD: log error?
+            continue
+        except KeyError:
+            continue
+        else:
+            item.save(update_fields=["data"])
+
+
+def migrate_schedule_logs(model, converter):
+    for item in model.objects.filter(
+        action__in=["wagtail.publish.schedule", "wagtail.schedule.cancel"]
+    ).only("data"):
```

Comment:
Use `.iterator()` in case of really big websites with lots of logs?

Reply:
Nice catch, I think I've used both in a project, but I will look into this tomorrow 👍

Reply:
fixed